regimes/mx/invoice_validator.go

```diff
@@ -30,9 +37,23 @@ func (v *invoiceValidator) validate() error {
 			),
 			validation.Skip, // Prevents each line's `ValidateWithContext` function from being called again.
 		),
+		validation.Field(&inv.Payment,
+			validation.Required,
+			validation.By(v.validPayment),
+		),
 	)
 }
 
+func (v *invoiceValidator) validateScenarios() error {
+	ss := v.inv.ScenarioSummary()
+
+	if ss.Codes[KeySATUsoCFDI] == "" {
```

Comment:
This feels a bit strange, especially the resulting error message as its not obvious what is missing. Perhaps the objective here is just to make sure that the invoice has a matching scenario and we can validate that instead?

Reply:
Yes, this one was a bit tricky. 

I borrowed the idea of validating scenarios from [the implementation in Italy](https://github.com/invopop/gobl/blob/main/regimes/it/invoice_validator.go#L125-L139). There, the error messages are like "missing scenario related to TipoDocumento" which I found that a GOBL user may find hard to decipher as scenarios are an internal, undocumented concept they don't need to be familiar with.

So I tried to come up with an error message that tells the users what they need do to fix the GOBL document using only concepts they can understand. Maybe a better wording would have been: "invoice must include a tax tag that maps to a UsoCFDI code". WDYT?

Reply:
Perhaps the message should just be `"'use' tax tag is required"`? But I agree the Italian example isn't very clear either, whatever we decide here should be reapplied there.

Reply:
Solved in https://github.com/invopop/gobl/pull/158/commits/c44f03520ad0bd823b058b0abd9a61cea8502231

Reply:
Although, in the case of the Italian cases, I don't think we can do like here. Both the invoice type and the tax tags tags may be very well present and the combination of them not map to any scenario. So, the best I think we could do there is something along the lines of my previous suggestion: "invoice type and tax tags doesn't map to a TipoDocumento code"